src/main/kotlin/net/lumalyte/lg/infrastructure/services/BankServiceBukkit.kt

```
@@ -173,58 +184,52 @@ class BankServiceBukkit(
+                    actorId = playerId,
+                    action = AuditAction.INSUFFICIENT_FUNDS,
+                    details = "Failed to withdraw money from player account"
+                ))
+                return null
+            }
 
+            // Credit the unified guild balance (store B: vault gold). This is atomic and the
+            // balance is immediately visible to all readers via VaultInventoryManager.
+            val creditedBalance = try {
+                vaultInventoryManager.depositGold(guildId, playerId, amount.toLong())
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)